router/Route.js

```diff
@@ -341,11 +341,30 @@ export default class Route {
 	_substituteOptionalParamInPath(path, paramName, paramValue) {
 		return path.replace(
 			new RegExp(`(^|\/):\\\?${paramName}([\/\?]|$)`),
-			paramValue ? '$1' + paramValue +  '$2' : ''
+			paramValue ? '$1' + this._encodePathParamValue(paramValue) +  '$2' : ''
 		);
 	}
 
 	/**
+	 * Encode special characters of path parameter value.
+	 * 
+	 * @param {string|number} paramValue Path parameter value
+	 * @return {string} Encoded path parameter value.
+	 */
+	_encodePathParamValue(paramValue) {
```

Comment:
@jurca It was request from our SEO master. After next discussion he changed his mind. So I am going to encode whole `paramValue`.